sycl/include/sycl/ext/oneapi/group_sort.hpp

```diff
@@ -0,0 +1,159 @@
+//==--------- group_sort.hpp --- SYCL extension group sorting algorithm-----==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+#include <CL/sycl/detail/defines_elementary.hpp>
+#include <CL/sycl/detail/group_sort_impl.hpp>
+#include <CL/sycl/detail/group_sort_util.hpp>
+#include <CL/sycl/detail/type_traits.hpp>
+#include <type_traits>
+
+#include "experimental/group_helpers_sorters.hpp"
+
+// define feature macro for group sorting algorithms
+#define SYCL_EXT_ONEAPI_GROUP_SORT 1
+
+__SYCL_INLINE_NAMESPACE(cl) {
+namespace sycl {
+namespace ext {
+namespace oneapi {
+namespace detail {
+
+// ---- traits
+template <typename T, typename = void> struct has_difference_type {};
+
+template <typename T>
+struct has_difference_type<T,
+#if __cplusplus < 201703
+                           cl::sycl::detail::void_type<
+#else
+                           std::void_t<
+#endif
```

Comment:
Nit. Did clang-format do that? Can we have some alias to avoid it?

Reply:
Since clang-format is passed, yep, this is its work. I've found this one in the repo: https://github.com/intel/llvm/blob/sycl/sycl/include/CL/sycl/detail/stl_type_traits.hpp#L42.

I'm just curious. What does "Nit" mean? I couldn't quickly find what is it.

Reply:
Nit means nitpick